File: trunk/cloud/android/common/src/main/java/org/openmobster/core/mobileCloud/android/util/XMLUtil.java
/**
 * Copyright (c) {2003,2011} {openmobster@gmail.com} {individual contributors as indicated by the @authors tag}.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */

package org.openmobster.core.mobileCloud.android.util;

/**
 * @author openmobster@gmail.com
 *
 */
public final class XMLUtil 
{	
	public static String cleanupXML(String xml)
	{
		String cleanedup = "";
		
		if(xml != null)
		{
			cleanedup = StringUtil.replaceAll(xml, "&", "&amp;");
			cleanedup = StringUtil.replaceAll(cleanedup,"<", "&lt;");
			cleanedup = StringUtil.replaceAll(cleanedup,">", "&gt;");			
			cleanedup = StringUtil.replaceAll(cleanedup,"\"", "&quot;");
			cleanedup = StringUtil.replaceAll(cleanedup,"'", "&apos;");
		}
		
		return cleanedup;
	}
		
	public static String restoreXML(String xml)
	{
		String restored = "";
		
		if(xml != null)
		{
			restored = StringUtil.replaceAll(xml,"&apos;", "'");
			restored = StringUtil.replaceAll(restored,"&quot;", "\"");
			restored = StringUtil.replaceAll(restored,"&gt;", ">");			
			restored = StringUtil.replaceAll(restored,"&lt;", "<");
			restored = StringUtil.replaceAll(restored,"&amp;", "&");
		}
		
		return restored;
	}
					
	public static String removeCData(String xml)
	{
		String cleanXML = StringUtil.replaceAll(xml,"<![CDATA[", "");
		cleanXML = StringUtil.replaceAll(cleanXML,"]]>", "");
		
		return cleanXML;
	}
		
	public static String addCData(String xml)
	{
		String cdataXml = "<![CDATA[";
		cdataXml += xml;
		cdataXml += "]]>";
		return cdataXml;
	}	
}

File: trunk/iphone/mobilecloudlib/XMLUtil.m
/**
 * Copyright (c) {2003,2010} {openmobster@gmail.com} {individual contributors as indicated by the @authors tag}.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */

#import "XMLUtil.h"


/**
 * 
 * @author openmobster@gmail.com
 */
@implementation XMLUtil

+(NSString *) cleanupXML:(NSString *)xml
{
	if([StringUtil isEmpty:xml])
	{
		return @"";
	}
	
	NSMutableString *cleanXml = [NSMutableString stringWithString:xml];
	
	NSRange searchRange = NSMakeRange(0, [cleanXml length]);
	[cleanXml replaceOccurrencesOfString:@"&" withString:@"&amp;" options:NSLiteralSearch range:searchRange];
	
	searchRange = NSMakeRange(0, [cleanXml length]);
	[cleanXml replaceOccurrencesOfString:@"<" withString:@"&lt;" options:NSLiteralSearch range:searchRange];
	
	searchRange = NSMakeRange(0, [cleanXml length]);
	[cleanXml replaceOccurrencesOfString:@">" withString:@"&gt;" options:NSLiteralSearch range:searchRange];
	
	searchRange = NSMakeRange(0, [cleanXml length]);
	[cleanXml replaceOccurrencesOfString:@"\"" withString:@"&quot;" options:NSLiteralSearch range:searchRange];

	searchRange = NSMakeRange(0, [cleanXml length]);
	[cleanXml replaceOccurrencesOfString:@"'" withString:@"&apos;" options:NSLiteralSearch range:searchRange];

	return cleanXml;
}

+(NSString *) restoreXML:(NSString *)xml
{
	if([StringUtil isEmpty:xml])
	{
		return @"";
	}
	
	NSMutableString *restoreXml = [NSMutableString stringWithString:xml];
	
	NSRange searchRange = NSMakeRange(0, [restoreXml length]);
	[restoreXml replaceOccurrencesOfString:@"&apos;" withString:@"'" options:NSLiteralSearch range:searchRange];
	
	searchRange = NSMakeRange(0, [restoreXml length]);
	[restoreXml replaceOccurrencesOfString:@"&quot;" withString:@"\"" options:NSLiteralSearch range:searchRange];
	
	searchRange = NSMakeRange(0, [restoreXml length]);
	[restoreXml replaceOccurrencesOfString:@"&gt;" withString:@">" options:NSLiteralSearch range:searchRange];
	
	searchRange = NSMakeRange(0, [restoreXml length]);
	[restoreXml replaceOccurrencesOfString:@"&lt;" withString:@"<" options:NSLiteralSearch range:searchRange];
	
	searchRange = NSMakeRange(0, [restoreXml length]);
	[restoreXml replaceOccurrencesOfString:@"&amp;" withString:@"&" options:NSLiteralSearch range:searchRange];
	
	return restoreXml;
}

+(NSString *) removeCData:(NSString *) xml
{
	NSMutableString *withOut = [NSMutableString stringWithString:xml];
	
	NSRange searchRange = NSMakeRange(0, [withOut length]);
	[withOut replaceOccurrencesOfString:@"<![CDATA[" withString:@"" options:NSLiteralSearch range:searchRange];
	
	searchRange = NSMakeRange(0, [withOut length]);
	[withOut replaceOccurrencesOfString:@"]]>" withString:@"" options:NSLiteralSearch range:searchRange];
	
	return withOut;
}

+(NSString *) addCData:(NSString *) xml
{
	NSMutableString *withCData = [NSMutableString stringWithString:@"<![CDATA["];
	
	[withCData appendString:xml];
	[withCData appendString:@"]]>"];
	
	return withCData;
}

@end
